src/cmd/singularity/cli/push.go

```diff
@@ -32,15 +31,11 @@ var PushCmd = &cobra.Command{
 	Args:   cobra.ExactArgs(2),
 	PreRun: sylabsToken,
 	Run: func(cmd *cobra.Command, args []string) {
-		if authWarning != auth.WarningEmptyToken &&
-			authWarning != auth.WarningTokenToolong &&
-			authWarning != auth.WarningTokenTooShort {
-			if authToken != "" {
-				libexec.PushImage(args[0], args[1], PushLibraryURI, authToken)
-				return
-			}
+		if authWarning == "" || authToken != "" {
+			libexec.PushImage(args[0], args[1], PushLibraryURI, authToken)
+		} else {
+			sylog.Fatalf("Couldn't push image to library: %v", authWarning)
```

Comment:
This still reads weirdly to me. Are we certain we cannot we get a situation where the authWarning is "" and token is also ""?

Reply:
I think the above logic should be `authWarning == "" && authToken != ""` or just `authWarning == ""`. I'm still a little confused about whether `authWarning` is meant to prevent usage of `authToken` or just warn about it. If there are cases where we want to use the token when a warning is also present, I'd suggest naming those (rather than the failure cases.) Missing a warning that should have been fatal was what caused the segmentation fault in the build code.